iceoryx_examples/request_response_in_c/client_c_basic.c

```diff
@@ -0,0 +1,128 @@
+const char APP_NAME[] = "iox-c-request-response-client-basic";
+
+void sigHandler(int signalValue)
+{
+    (void)signalValue;
+    keepRunning = false;
+}
+
+int main()
+{
+    //! [register signal handler and init runtime]
+    signal(SIGINT, sigHandler);
+    signal(SIGTERM, sigHandler);
+
+    iox_runtime_init(APP_NAME);
+    //! [register signal handler and init runtime]
+
+    //! [create client]
+    iox_client_storage_t clientStorage;
+    iox_client_t client = iox_client_init(&clientStorage, "Example", "Request-Response", "Add", NULL);
+    //! [create client]
+
+    //! [define variables]
+    uint64_t fibonacciLast = 0U;
+    uint64_t fibonacciCurrent = 1U;
+    int64_t requestSequenceId = 0;
+    int64_t expectedResponseSequenceId = requestSequenceId;
+    //! [define variables]
+
+    //! [main loop]
+    while (keepRunning)
+    {
+        //! [loan request]
+        struct AddRequest* request = NULL;
+        enum iox_AllocationResult loanResult =
+            iox_client_loan_request(client, (void**)&request, sizeof(struct AddRequest));
+        //! [loan request]
+
+        if (loanResult == AllocationResult_SUCCESS)
+        {
+            //! [set sequence id]
+            iox_request_header_t requestHeader = iox_request_header_from_payload(request);
+            iox_request_header_set_sequence_id(requestHeader, requestSequenceId);
+            expectedResponseSequenceId = requestSequenceId;
+            requestSequenceId += 1;
+            //! [set sequence id]
+
+            //! [set and send request]
+            request->augend = fibonacciLast;
+            request->addend = fibonacciCurrent;
+            printf("%s Send Request: %lu + %lu\n",
+                   APP_NAME,
+                   (unsigned long)fibonacciLast,
+                   (unsigned long)fibonacciCurrent);
+            iox_client_send(client, request);
+            //! [set and send request]
+
+            //! [wait for response]
+            const uint32_t DELAY_TIME_IN_MS = 150U;
+            sleep_for(DELAY_TIME_IN_MS);
+            //! [wait for response]
+
+            //! [process response]
+            const struct AddResponse* response = NULL;
+            while (iox_client_take_response(client, (const void**)&response) == ChunkReceiveResult_SUCCESS)
+            {
+                iox_const_response_header_t responseHeader = iox_response_header_from_payload_const(response);
+                int64_t receivedSequenceId = iox_response_header_get_sequence_id_const(responseHeader);
+                if (receivedSequenceId == expectedResponseSequenceId)
+                {
+                    fibonacciLast = fibonacciCurrent;
+                    fibonacciCurrent = response->sum;
+                    printf("%s Got Response: %lu\n", APP_NAME, (unsigned long)fibonacciCurrent);
+                }
+                else
+                {
+                    printf("Got Response with outdated sequence ID! Expected = %lu; Actual = %lu! -> skip\n",
```

Comment:
Could this even happen if we would wait (spin) for the response? (unless there is a bug, of course).

Reply:
In theory it should not happen, but this is an example which should demonstrate every facet - this includes error handling. The users can then decide for themselves if they want to omit that.

Reply:
It can also practically happen when the server is too slow. The example waits for 150ms for a response. If the server is also polling with 1000ms there would be and outdated response when the queue is polled the next time